renderjson.js

```diff
@@ -97,17 +97,26 @@ var module, window, define, renderjson=(function() {
                                                    a.onclick = function(e) { callback(); if (e) e.stopPropagation(); return false; };
                                                    return a; };
 
+    var walkDOM = function(node, func){
```

Comment:
This is an interesting function, and use case.  But not all json object graphs are tree.  We explicitly, turn the tree back into a graph (for complex DTO object graphs and json schema), before rendering them. 
https://github.com/andifreed/renderjson/blob/50a85654206671137e397d9d92996a94f4e9b06a/resolvejson.js

But there may be other graphs passed to this renderer, you can make it impervious to such infinite recursion by adding a array|stack parameter and checking that the object hasn't been seen yet before descending.  Besure to put before descending and pop after returning.